### ISSUE TITLE
`onSuccess` not called in `SupaSocialsAuth`

### ISSUE THREAD

**Issue by dannypurcell** [supabase-community/flutter-auth-ui#70]
# Bug report

## Describe the bug

`onSuccess` not called in `SupaSocialsAuth`

Accoding to the readme https://github.com/supabase-community/flutter-auth-ui#social-auth
this was intended to work as the other auth flows do.

## To Reproduce

* Set a function to `SupaSocialsAuth(onSuccess)`
* Set a breakpoint or print something in the function body.
* Run login flow
* Breakpoint/print is never triggered.

## Expected behavior

`onSuccess` should be called in the case of `SupaSocialsAuth` as it is with the other auth flows.

## Additional context

This appears to be due to  `signInWithOAuth` being implemented as an extension to `GoTrueClient` and therefore not being able to call `notifyAllSubscribers`. The `pkce` flow to the external provider is also kind of opaque from the perspective of `signInWithOAuth`, we're just launching the url so there's not a great way to know for sure that it succeeded from that point in the code.

**Comment by dshukertjr** [comment]
The `onSuccess` handler should get called after you complete the authentication flow and come back to the app as long as you go back to the same page. 

If anything is not working, could you provide more details on how to reproduce the issue? What platform are you running the app on? Do you have an active session after completing the OAuth login flow and not see a `onSuccess` handler being called, or are you not signed in at all after completing the OAuth flow?

**Comment by dannypurcell** [comment]
Yea, no problem!

Platform is `Flutter/Dart` and I do have an active session after completing. No trouble grabbing the session information with `supabase.auth.onAuthStateChange.listen`, it's just `onSuccess` is never called.

**Comment by dshukertjr** [comment]
@dannypurcell 
Thanks for the info, and I wanted to ask what platform as in it is on Android, iOS, web or desktop.

**Comment by dannypurcell** [comment]
Ah, my build targets are all three plus Windows and Linux, but in this case I've been testing on Web.

And I think I see what you're getting at there.

Looking in to the code flow further, I think what may be happening is just that the `SocialsAuth` widget is not mounted at the point in time when `Supabase.instance.client.auth.onAuthStateChange` has a session.

When I first logged this issue I was under the impression that the `/authorize` call was somehow "out of band" as it were, that the Stream registration was still valid after we started the login prompt process, which would be the case if we were using a popup dialog, I think. At least on the Web platform though, I see the `launchUrl` call is targeting `_self` which is what makes the tab navigate off and therefore unload the app.

Once we have the auth `code` and get redirected back to the app domain, we're on a fresh load of the app so the subscription to the `onAuthStateChange` stream has been garbage collected already and my `GoRouter` redirect, which is set up to send the user to the `/login` route whenever `Supabase.instance.client.auth.currentSession` is null, sees that we have a session now so the SocialsAuth widget is not loaded in a context where that subscription would ever see the `initialSession` event.

I actually put that check in the router after logging this but I must have been doing something else to end up not loading the login widget again earlier, I've been doing a lot of experimenting and learning Supabase with this app.
